Controller/Annotations/AbstractScalarParam.php

```diff
@@ -65,7 +65,11 @@ public function getConstraints()
         }
 
         if (false === $this->allowBlank) {
-            $constraints[] = new NotBlank();
+            $notBlank = new NotBlank();
+            if (isset($notBlank->allowNull)) {
```

Comment:
Are you sure this works as expected? I would expect this to always evaluate to `false` (see also https://3v4l.org/sdhnT). Probably using `property_exists()` is what we need to use here.

Reply:
For test it, I have commented the $allowNull value of the Constraint class and I didn't get any new error, It worked like expected (still having the bug #1496, but not adding new exceptions). I can use the property_exists if it is cleaner so.

Reply:
If we didn't use `property_exists()`, our code would depend on the default value of the `$allowNull` property of the constraint class.

Reply:
Thank you very much for your recomendation @xabbuh, I've modified the condition according your advise. I believe that now it's fine.